karton/core/base.py

```diff
@@ -191,12 +192,24 @@ def __init__(
     ) -> None:
         super().__init__(config=config, identity=identity, backend=backend)
         self.setup_logger()
-        self.shutdown = False
-        self.killer = GracefulKiller(self.graceful_shutdown)
+        self._shutdown = False
 
-    def graceful_shutdown(self) -> None:
-        self.log.info("Gracefully shutting down!")
-        self.shutdown = True
+    def _do_shutdown(self) -> None:
+        self.log.info("Got signal, shutting down...")
+        self._shutdown = True
+
+    @property
+    def shutdown(self):
```

Comment:
Is there a chance that some users try to shut down the service on their own by setting this to True?

Reply:
I haven't seen that pattern, but we can expose it in more controlled way in a future.